scripts/classes/class.HttpConfigBase.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/***
+ * Class HttpConfigBase
+ *
+ * Base class for all HTTP server configs
+ *
+ */
+class HttpConfigBase {
+
+	/**
+	 * process special config as template, by substituting {VARIABLE} with the
+	 * respective value.
+	 *
+	 * The following variables are known at the moment:
+	 *
+	 * {DOMAIN}         - domain name
+	 * {IP}             - IP for this domain
+	 * {PORT}           - Port for this domain
+	 * {CUSTOMER}       - customer name
+	 * {IS_SSL}         - '1' if domain/ip is ssl, '' otherwise
+	 * {DOCROOT}        - document root for this domain
+	 *
+	 * @param $template
+	 * @return string
+	 */
+	protected function processSpecialConfigTemplate($template, $ipandport, $domain) {
+		$templateVars = array(
+			'DOMAIN' => $domain['domain'],
+			'CUSTOMER' => $domain['loginname'],
+			'IP' => $ipandport['ip'],
+			'PORT' => $ipandport['port'],
+			'IS_SSL' => $domain['ssl'],
```

Comment:
$domain['ssl'] won't be an empty string, it's either 1 or 0, also, whether it's a ssl-vhost should be read from whether the currently used ip/port is SSL=Yes.

And i don't get what a '1' for enabled SSL is useful for except for your very special case to distinguish between ssl and non-ssl as you cannot use IF's there